core/src/main/java/io/grpc/ClientInterceptors.java

```diff
@@ -204,4 +216,190 @@ public final void start(Listener<RespT> responseListener, Metadata headers) {
       }
     }
   }
+
+  public static class RetryingInterceptor implements ClientInterceptor {
+    // TODO(lukaszx0) have default logger for this one to give more visibility into when/what gets
+    // retried?
+    // private final Logger logger = Logger.getLogger(RetryingInterceptor.class.getName());
+    private final RetryPolicy retryPolicy;
+
+    public RetryingInterceptor(RetryPolicy retryPolicy) {
+      this.retryPolicy = retryPolicy;
+    }
+
+    public RetryingInterceptor() {
+      this.retryPolicy = new IdempotentRetryPolicy();
```

Comment:
You need a provider here, so you can create a new policy per call. Otherwise, the backoff is shared across all calls and cumulative for the whole system, instead of per-call.